src/etc/inc/openvpn.inc

```diff
@@ -1108,6 +1108,11 @@ function openvpn_reconfigure($mode, $settings) {
 			if (!($settings['auth_user'] && $settings['auth_pass'])) {
 				$userpass .= "\n";
 			}
+
+			// if auth_script is not empty, we use its output instead of user/pass
+			if ($settings['auth_script']) {
+				$userpass = shell_exec($settings['auth_script']);
```

Comment:
This needs a test to check that the script exists before it is executed, otherwise you are blindly sending whatever the user entered to run as a shell command.